src/lib/crypto/kimchi_bindings/stubs/src/caml/caml_pointer.rs

```diff
@@ -49,16 +51,15 @@ macro_rules! impl_caml_pointer {
                 unsafe {
                     // Wholely unsafe, Batman!
                     // We would use [`get_mut_unchecked`] here, but it is nightly-only.
-                    // Instead, we get coerce our constant pointer to a mutable
-                    // pointer, in the knowledge that
+                    // Instead, we use UnsafeCell in the knowledge that
                     // * all of our mutations called from OCaml are blocking, so
-                    // we won't have multiple live mutable references live
+                    // we won't have multiple live mutable references
                     // simultaneously, and
                     // * the underlying pointer is in the correct state to be
                     //   mutable, since we can call [`get_mut_unchecked`] in
                     //   nightly, or can call [`get_mut`] and unwrap if this is
                     //   the only live reference.
-                    &mut *(((&*self.0) as *const Self::Target) as *mut Self::Target)
```

Comment:
`1.74` throws a compiler error here. There might be some other way to get it to stop complaining, but I think this is equivalent without the illegal cast